rest/session_api.go

```diff
@@ -182,13 +182,12 @@ func (h *handler) makeSessionFromNameAndEmail(username, email string, createUser
 			if err = user.SetEmail(email); err == nil {
 				h.db.Authenticator().Save(user)
 			}
-		}
-
-		// User not found by username. Attempt user lookup by email. This provides backward
-		// compatibility for users that were originally created with id = email
-		user, err = h.db.Authenticator().GetUserByEmail(email)
-		if err != nil {
-			return err
+		} else {
```

Comment:
I think this needs to be `else if user == nil`.  Otherwise we'll attempt to re-lookup the user when the user's found and the email matches.

Reply:
Thanks, my attempt at reducing the nesting obviously impacted my ability to parse it well. Restored original code, and wrapped in the extra if.